### PR TITLE
Add ChainActionListNode to chain a list of Actions

### PR DESCRIPTION
**Pull request by MagicMau**
While using a loop to generate a list of Actions, I was missing a way to chain the list into a single Action. I have modified the existing ChainAction to allow a list as input to handle this use case.

### CODE REVIEW

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    bl_label = "Chain Action List"
+    errorHandlingType = "EXCEPTION"
+
+    __annotations__ = {}
+    __annotations__["relative"] = BoolProperty(name = "Relative", default = True,
+        description = "Use last frame of first action as starting point of the second",
+        update = propertyChanged)
+
+    def create(self):
+        self.newInput("Action List", "Action list", "list")
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Use more explicit names, `list` -> `actions`, `r` -> `chainedAction`, and so on.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        self.newOutput("Action", "Chained Action")
+
+    def draw(self, layout):
+        layout.prop(self, "relative")
+
+    def execute(self, list):
+        if list is None:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: I don't think `list` can ever be `None`, so this is redundant.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        if length == 0:
+            return None
+
+        if length < 2:
+            return list[0]
+        
+        r = list[0]
+
+        if not isinstance(r, BoundedAction):
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Check the whole list upfront.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            self.raiseErrorMessage("only bounded actions can be chained")
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Capitalize and end with a period.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+        i = 1
+        while i < length:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Use a `for` loop instead.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            b = list[i]
+            if isinstance(b, BoundedAction):
+                r = ChainAction(r, b, self.relative)
+            i += 1
+
+        return r
+
+cdef class ChainAction(BoundedAction):
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Aren't those classes identical to those in the Chain Actions node? Can't we just use the already defined classes?

File: animation_nodes/ui/node_menu.py [hunk 7]
```diff
@@ -472,6 +472,8 @@ def draw(self, context):
         insertNode(layout, "an_WiggleActionNode", "Wiggle Action")
         insertNode(layout, "an_FollowSplineActionNode", "Follow Spline Action")
         layout.separator()
+        insertNode(layout, "an_ChainActionListNode", "Chain Action List")
```
Comment (OmarEmaraDev) [inline animation_nodes/ui/node_menu.py]: Wouldn't it be better to put that below the *Chain Actions* node?

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+)
+
+class ChainActionListNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_ChainActionListNode"
+    bl_label = "Chain Action List"
+    errorHandlingType = "EXCEPTION"
+
+    __annotations__ = {}
+    __annotations__["relative"] = BoolProperty(name = "Relative", default = True,
+        description = "Use last frame of first action as starting point of the second",
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Update the description to fit the fact that we are dealing with a list.

File: animation_nodes/nodes/action/chain_action_list.pyx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    def execute(self, list):
+        if list is None:
+            return list
+
+        length = len(list)
+
+        if length == 0:
+            return None
+
+        if length < 2:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/action/chain_action_list.pyx]: Make this `if length == 1`, just to more readable.

**Comment by JacquesLucke** [comment]
Looks like this was closed automatically because I merged blender2.8 into master. Also I cannot reopen it. Please create a new pull request that targets master, thank you.

**Pull request closed without merging**